src/Adapter/Manufacturer/QueryHandler/GetManufacturerForEditingHandler.php

```diff
@@ -0,0 +1,105 @@
+<?php
+
+/**
+ * 2007-2018 PrestaShop
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2018 PrestaShop SA
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\PrestaShop\Adapter\Manufacturer\QueryHandler;
+
+use Manufacturer;
+use PrestaShop\PrestaShop\Core\Domain\Manufacturer\EditableManufacturer;
+use PrestaShop\PrestaShop\Core\Domain\Manufacturer\Exception\ManufacturerException;
+use PrestaShop\PrestaShop\Core\Domain\Manufacturer\Exception\ManufacturerNotFoundException;
+use PrestaShop\PrestaShop\Core\Domain\Manufacturer\Query\GetManufacturerForEditing;
+use PrestaShop\PrestaShop\Core\Domain\Manufacturer\Query\Handler\GetManufacturerForEditingInterface;
+use PrestaShop\PrestaShop\Core\Domain\Manufacturer\VO\ManufacturerId;
+
+/**
+ * Looks up a manufacturer for editing
+ */
+class GetManufacturerForEditingHandler implements GetManufacturerForEditingInterface
+{
+
+    /**
+     * @param GetManufacturerForEditing $query
+     *
+     * @return EditableManufacturer
+     * @throws ManufacturerNotFoundException
+     * @throws ManufacturerException
+     */
+    public function handle(GetManufacturerForEditing $query)
+    {
+        $entity = $this->loadById($query->getManufacturerId());
+
+        return $this->buildEditableManufacturer($entity);
+    }
+
+    /**
+     * @param ManufacturerId $manufacturerId
+     *
+     * @return Manufacturer
+     * @throws ManufacturerNotFoundException
+     */
+    private function loadById(ManufacturerId $manufacturerId)
```

Comment:
Seems like a good method to be extracted to a repository. You will probably be needing in a lot of other handlers too.

Reply:
Of course, but the problem is we don't have entities to load from a repository right now, as the ObjectModel is currently both the entity and the repository. This should be of course refactored to have proper models on the next major version.